experimental/big_paper/plots.ipynb

```diff
@@ -1,1147 +1,1365 @@
+   "source": [
+    "from google.colab import auth\n",
+    "auth.authenticate_user()\n",
+    "\n",
+    "project_id = 'marginalization-external-xgcp'\n",
+    "!gcloud config set project {project_id}\n",
+    "!gsutil cp gs://ub-checkpoints/big-paper-raw-measurements.pkl /tmp/big-paper-raw-measurements.pkl\n",
+    "path = '/tmp/big-paper-raw-measurements.pkl'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "JcnAzeyWsHu8"
+   },
+   "outputs": [],
+   "source": [
+    "with tf.io.gfile.GFile(path, 'rb') as f:\n",
+    "  raw_measurements = pickle.load(f)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# Split RETINA results into the two distributional shifts: Country Shift and Severity Shift\n",
+    "\n",
+    "shift_map = {'aptos': 'country', 'severity': 'severity'}\n",
+    "\n",
+    "def add_distribution_shift_to_retina_ds_name(row):\n",
+    "  dataset = str(row['config.dataset'])\n",
+    "  if dataset == 'retina':\n",
+    "    shift = shift_map[str(row['config.distribution_shift'])]\n",
+    "    row['config.dataset'] = f'{dataset}_{shift}'\n",
+    "\n",
+    "  return row\n",
+    "\n",
+    "def split_retina_results_by_shifts(raw_dict):\n",
+    "  for model in raw_dict.keys():\n",
+    "    raw_model_df = raw_dict[model]\n",
+    "    if not len(raw_model_df[raw_model_df['config.dataset'] == 'retina']):\n",
+    "        continue\n",
+    "    \n",
+    "    print(f'Splitting RETINA results for model {model} by distribution shift.')\n",
+    "\n",
+    "    raw_model_df = raw_model_df.apply(\n",
+    "        add_distribution_shift_to_retina_ds_name, axis='columns')\n",
+    "    raw_dict[model] = raw_model_df\n",
+    "\n",
+    "  return raw_dict\n",
+    "\n",
+    "raw_measurements = split_retina_results_by_shifts(raw_measurements)\n"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "K6Bc5TFN3ZTk"
+   },
+   "outputs": [],
+   "source": [
+    "dataset_metric = {\n",
+    "    'cifar10': 'val_loss',\n",
+    "    'cifar100': 'val_loss',\n",
+    "    'imagenet2012': 'val_loss',\n",
+    "    'imagenet21k': 'val_loss',\n",
+    "    'jft/entity:1.0.0': 'val_loss',\n",
+    "    'retina_country': 'in_domain_validation/auroc',\n",
+    "    'retina_severity': 'in_domain_validation/auroc'\n",
+    "}\n",
+    "measurements = get_optimal_results(raw_measurements,\n",
+    "                                   dataset_metric=dataset_metric)\n",
+    "\n",
+    "df = preprocess(measurements)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Add RETINA ImageNet-21K Results"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "! pip install wandb"
```

Comment:
This code relies on importing the code from wandb. We don't have access to that with your Google machines. It's good that the #s can be loaded programmatically though. Similar to our XManager exporting, how about having a code block that loads from wandb and saves it to disk? That way, folks just need to load a DF into memory.